app/src/main/java/com/nmakademija/nmaakademija/fragment/OnboardingFragment.java

```diff
@@ -0,0 +1,95 @@
+package com.nmakademija.nmaakademija.fragment;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v4.app.Fragment;
+import android.support.v7.preference.PreferenceManager;
+import android.support.v7.widget.DefaultItemAnimator;
+import android.support.v7.widget.LinearLayoutCompat;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.nmakademija.nmaakademija.MainActivity;
+import com.nmakademija.nmaakademija.R;
+import com.nmakademija.nmaakademija.adapter.DividerItemDecoration;
+import com.nmakademija.nmaakademija.adapter.SectionsAdapter;
+import com.nmakademija.nmaakademija.api.API;
+import com.nmakademija.nmaakademija.entity.Section;
+import com.nmakademija.nmaakademija.listener.ClickListener;
+import com.nmakademija.nmaakademija.listener.RecyclerTouchListener;
+import com.nmakademija.nmaakademija.utils.Error;
+
+import java.util.List;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class OnboardingFragment extends Fragment {
+
+    private boolean isFirstTime;
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater,
+                             @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_onboarding, container, false);
+    }
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+
+        isFirstTime = PreferenceManager.getDefaultSharedPreferences(getContext().getApplicationContext())
+                .getString(getString(R.string.section_key), "0")
```

Comment:
Too many duplicated logic related to preferences. Move all preferences related things to separate class, have functions `isFirstTime`,`setSection` and etc.